src/dxvk/dxvk_device_filter.cpp

```diff
@@ -1,13 +1,25 @@
 #include "dxvk_device_filter.h"
 
+std::string convertUUID(const uint8_t* uuid) {
```

Comment:
Doesn't seem to match how UUIDs are commonly formatted at all?

Reply:
Thanks for taking a look at this. Do you have a reference that I can look at so that I can fix the implementation? As far as I was aware, UUID's were simply a string of chars, so I assumed this would be an ok, albeit ugly solution.

Reply:
From [wikipedia](https://en.m.wikipedia.org/wiki/Universally_unique_identifier#Format):

> In its canonical textual representation, the 16 [octets](https://en.m.wikipedia.org/wiki/Octet_(computing)) of a UUID are represented as 32 [hexadecimal](https://en.m.wikipedia.org/wiki/Hexadecimal) (base-16) digits, displayed in five groups separated by hyphens, in the form 8-4-4-4-12 for a total of 36 characters (32 hexadecimal characters and 4 hyphens). For example:
> 
>     123e4567-e89b-12d3-a456-426614174000
>     xxxxxxxx-xxxx-Mxxx-Nxxx-xxxxxxxxxxxx

I'm not aware of a format for UUIDs that only uses the digits 0-9, but that doesn't mean it doesn't exist. (i suspect it would be along the lines of interpreting the uuid as a single 128 bit number to avoid ambiguity, though.)

Reply:
@meladath 
Somewhat of a related UUID issue from CUDA: [https://stackoverflow.com/questions/68823023/set-cuda-device-by-uuid](https://stackoverflow.com/questions/68823023/set-cuda-device-by-uuid)

Sample code snippit there to select adapter based on UUID.

So, since it is octets/base-16 the number: 3f12  is not the same as 3f 12 😄 I have seen uuid/luid being treated like a char array - `char uuid[16]`

Reply:
We now use uint32_t, this will preserve any hex chars and allow UUID's to correctly be captured and compared.

Reply:
[https://github.com/KhronosGroup/Vulkan-Tools/blob/master/vulkaninfo/outputprinter.h#L35-L43](https://github.com/KhronosGroup/Vulkan-Tools/blob/master/vulkaninfo/outputprinter.h#L35-L43)
Is that not the function Vulkaninfo use to output this UUID?

I am not sure that uint32_t would "preserve" the hex numbers any better... char == uint8_t != uint32_t.

But tbh, i am by no means any authority on that matter.

Reply:
Hey, thats correct, I was looking at some old source where it simply cast each array element into a uint32_t and output it to cout using a formatting string which added the - chars in.

You are correct about it not being any better to preserve data than a char, the problem was in my original implementation I was casting it to an int, resulting in erroneous values for UUID's that contains any character from a-f.